WordPress/Classes/ViewRelated/Blog/BlogDetailsViewController.m

```diff
@@ -312,13 +312,13 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
 {
     [tableView deselectRowAtIndexPath:indexPath animated:YES];
 
-    if ([self isConfigurationSection:indexPath.section] && indexPath.row == BlogDetailsRowEditSite) {
-        SiteSettingsViewController *editSiteViewController = [[SiteSettingsViewController alloc] initWithBlog:self.blog];
-        [self.navigationController pushViewController:editSiteViewController animated:YES];
-    }
-
-    Class controllerClass;
-    if ([self isGeneralSection:indexPath.section]) {
+    if ([self isConfigurationSection:indexPath.section]) {
+        switch (indexPath.row) {
+            case BlogDetailsRowEditSite:
+                [self showSettingsForBlog:self.blog];
+                break;
+        }
```

Comment:
I guess it makes sense for consistency with the rest of the conditionals to have a nested switch with a single case, but I'm a little worried that this change doesn't handle a situation where `indexPath.row != BlogDetailsRowEditSite`. Maybe a default clause with an assert / log to at least flag that potential case?